source/common/router/rds_impl.h

```diff
@@ -145,22 +147,29 @@ class RdsRouteConfigSubscription : Envoy::Config::SubscriptionCallbacks,
   RdsRouteConfigSubscription(
       const envoy::extensions::filters::network::http_connection_manager::v3::Rds& rds,
       const uint64_t manager_identifier,
-      Server::Configuration::ServerFactoryContext& factory_context,
-      ProtobufMessage::ValidationVisitor& validator, Init::Manager& init_manager,
-      const std::string& stat_prefix,
+      Server::Configuration::ServerFactoryContext& factory_context, const std::string& stat_prefix,
       RouteConfigProviderManagerImpl& route_config_provider_manager);
 
   bool validateUpdateSize(int num_resources);
   static std::string loadTypeUrl(envoy::config::core::v3::ApiVersion resource_api_version);
 
-  Init::Manager& getRdsConfigInitManager() { return init_manager_; }
-
   std::unique_ptr<Envoy::Config::Subscription> subscription_;
   const std::string route_config_name_;
   Server::Configuration::ServerFactoryContext& factory_context_;
   ProtobufMessage::ValidationVisitor& validator_;
-  Init::Manager& init_manager_;
-  Init::SharedTargetImpl init_target_;
+
+  // Init target used to notify the parent init manager that the subscription [and its sub resource]
+  // is ready.
+  Init::SharedTargetImpl parent_init_target_;
+
+  // Init watcher on RDS and VHDS ready event. This watcher marks parent_init_target_ ready.
+  Init::WatcherImpl init_watcher_;
```

Comment:
Might want to call this `parent_init_watcher_` or `local_init_watcher_` (I'm not sure which) for consistency.

Reply:
moved to config-provider-framework now. named to local_init_watcher_.

Reply:
> latest init manager related changes still lgtm

Thanks!